Use NodeIPAMController code from K8S (#1561)
antrea-io/antrea: docs/feature-gates.md

| `NetworkPolicyStats`    | Agent + Controller | `true`  | Beta  | v0.10         | v1.2         | N/A        | No                 |       |
| `NodePortLocal`         | Agent              | `false` | Alpha | v0.13         | N/A          | N/A        | Yes                |       |
| `Egress`                | Agent + Controller | `false` | Alpha | v1.0          | N/A          | N/A        | Yes                |       |

> | `NodeIPAMController`    | Controller         | `false` | Alpha | v1.4          | N/A          | N/A        | Yes                |       |

Reviewer: I thought we were trying to merge this in time for v1.3, bit v1.4 works for me since I'm late with my review :)

cc @vicky-liu

Reviewer 2: We might need a pre-release build for the uses case the will need to leverage this feature

This feature is currently only supported for Nodes running Linux and "encap"
mode. The support for Windows and other traffic modes will be added in the
future.

### NodeIPAMController

`NodeIPAMController` runs Kubernetes' Node IPAM Controller within the Antrea
Controller. Running Node IPAM Controller with Antrea is useful in environments
where Kubernetes Controller Manager does not run the Node IPAM Controller, and
Antrea has to handle the CIDR allocation.

#### Requirements for this Feature

> This feature requires that Node IPAM Controller will be disabled in Kunbernetes

Reviewer: ```suggestion
This feature requires the Node IPAM Controller to be disabled in Kubernetes
```
